src/SurveyDesign.jl

```diff
@@ -78,20 +78,25 @@ struct SurveyDesign <: AbstractSurveyDesign
         sampsize_labels = :sampsize
         data[!, sampsize_labels] = fill(length(unique(data[!, cluster])), (nrow(data),))
         if !(typeof(popsize) <: Nothing)
-            data[!, :weights] = data[!, popsize] ./ data[!, sampsize_labels]
-        elseif !(typeof(weights) <: Nothing)
-            data.weights = data[!, weights]
+            weights_labels = :weights
+            data[!, weights_labels] = data[!, popsize] ./ data[!, sampsize_labels]
+        elseif typeof(weights) <: Symbol
+            if !(typeof(data[!, weights]) <: Vector{<:Real})
+                error(string("given weights column ", weights , " is not of numeric type"))
+            end
+            weights_labels = weights
         else
-            data.weights = repeat([1], nrow(data))
+            weights_labels = :weights
+            data[!, weights_labels] = repeat([1], nrow(data))
         end
-        data[!, :probs] = 1 ./ data[!, :weights] # Many formulae are easily defined in terms of sampling probabilties
-        data[!, :allprobs] = data[!, :probs] # In one-stage cluster sample, allprobs is just probs, no multiplication needed
-        pps = false
+        allprobs_labels = :allprobs
```

Comment:
Since in our implementation there is no difference between `probs` and `allprobs`, why not just keep the name `probs`? (Or `_probs`, following the above `_weights` argument.)

Reply:
Technically in the single stage approximation, `allprobs` is what we are actually using (so then `probs` should be dropped). 

We want SurveyDesign to be the most general design. We need the flexibility to add full multistage sampling with `fpc` and `probs` for each stage. So I dont see why we should remove `probs` right now.